### ISSUE TITLE
NumberField buttons trigger change when scrolling on touch devices

### ISSUE THREAD

**Issue by cbodin** [adobe/react-spectrum#8472]
### Provide a general summary of the issue here

When scrolling on touch devices on a page with a `<NumberField>` including increment and decrement buttons, it's very easy to accidentally trigger a change without noticing.

### 🤔 Expected Behavior?

A scrolling gesture should not trigger any actions as that is pretty unexpected and not default behavior on touch devices. Starting a scroll on the increment/decrement buttons should not do anything.

### 😯 Current Behavior

Starting the scroll on the increment/decrement buttons will trigger a change.

### 💁 Possible Solution

The default behavior should probably be to trigger change on `click` / `pointerup` / `touchend` instead of `pointerdown` / `touchstart` for touch devices or prevent the change action if the page is being scrolled.

### 🔦 Context

We are using the NumberField in multiple booking systems with almost all users on mobile devices. We are getting frequent user complaints that they are creating bookings with incorrect quantities.

After checking analytics and logs, we can see that they have indeed triggered the change via the buttons without noticing it.

The problem is probably much more common for us than what we get reports of, as we only hear from the user's that did not notice the incorrect quantity on confirmation pages. Those who notice can edit to the correct quantity before proceeding, thus never contacting support.

### 🖥️ Steps to Reproduce

This can be reproduced by scrolling on the increment/decrement buttons on the NumberField documentation page: https://react-spectrum.adobe.com/react-aria/NumberField.html

Increasing the button sizes and placing them on the right side to make it easier to click will also inadvertently make this issue pop up very often. 

<video src="https://github.com/user-attachments/assets/c866bb0e-6706-4229-93a6-50d09900b079"></video>
<video src="https://github.com/user-attachments/assets/78862207-75af-44f1-9816-daf76b4c8723"></video>

### Version

1.10.1

### What browsers are you seeing the problem on?

Chrome, Safari

### What operating system are you using?

MacOS, iOS, Android

**Comment by snowystinger** [comment]
I see it only happens with RAC, not the React Spectrum component, so this is probably fixed with some CSS in the docs and in your application.

I do see that it happens in the Chrome device emulator, but I'm not seeing on my real device. Will need someone else to verify with Android though.

Can you apply `touch-action: none` to your stepper buttons?
Like this [fix: RAC Numberfield scrolling with stepper buttons](https://github.com/adobe/react-spectrum/pull/8474#top)

**Comment by cbodin** [comment]
Testing the React Spectrum component example on a real device, i notice multiple problems now:
- Users can still accidentally trigger changes when starting a scrolling gesture.
- Users are unable to scroll if the gesture starts on one of the stepper buttons.
- Users are unable to pinch-to-zoom if the gesture starts on one of the stepper buttons.

This breaks expected browser behavior — especially when buttons are placed where users naturally scroll.

<video src="https://github.com/user-attachments/assets/849ed069-304e-4cde-a58a-990f8b14d28b"></video>

**Comment by devongovett** [comment]
One behavior that depends on triggering changes on pointer down is pressing and holding to repeatedly increment/decrement the value. We could:

* Disable that on touch devices and only do it on pointer up
* Add a prop to disable that
* Add a delay to trigger the repeating increment/decrement instead of triggering the first one instantly

**Comment by cbodin** [comment]
An `<input type="number">` will have increment/decrement buttons in Chrome per default. These trigger after a delay which seems resonable. The delay should probably be cancelled if the user starts scrolling.

A prop to completely disable the press & hold would be great as it's not a necessary or expected feature on fields with a small range, e.g. 1-10.
